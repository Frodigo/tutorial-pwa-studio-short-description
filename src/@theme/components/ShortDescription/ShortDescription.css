.root {
    margin: 15px 0;
}

.section {
    border-color: rgb(var(--venia-border));
    border-style: solid;
    border-width: 1px 0 1px;
    margin: 0 1.5rem;
    padding: 1.5rem 0;
}
