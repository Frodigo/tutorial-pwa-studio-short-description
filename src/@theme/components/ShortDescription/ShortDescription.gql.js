import gql from 'graphql-tag';

const GET_SHORT_DESCRIPTION_QUERY = gql`
    query shortDescriptionOfProduct($productSku: String!) {
        products(filter: { sku: { eq: $productSku } }) {
            items {
                uid
                short_description {
                    html
                }
            }
        }
    }
`;

export default {
    queries: {
        getShortDescriptionQuery: GET_SHORT_DESCRIPTION_QUERY
    },
    mutations: {}
};
