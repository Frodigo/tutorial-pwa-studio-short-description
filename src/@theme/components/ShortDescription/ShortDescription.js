import React, { useMemo } from 'react';
import { useQuery } from '@apollo/client';
import RichText from '@magento/venia-ui/lib/components/RichText';

import classes from './ShortDescription.css';
import productOperations from './ShortDescription.gql';

import { shape, string } from 'prop-types';

const ShortDescription = props => {
    const { productSku } = props;
    const { queries } = productOperations;
    const { getShortDescriptionQuery } = queries;

    const { data } = useQuery(getShortDescriptionQuery, {
        fetchPolicy: 'cache-and-network',
        variables: {
            productSku
        }
    });

    const shortDescription = useMemo(() => {
        if (!data) return null;

        const { products } = data;

        if (
            products &&
            products.items &&
            products.items.length &&
            products.items[0].short_description &&
            products.items[0].short_description.html &&
            products.items[0].short_description.html.length
        ) {
            return products.items[0].short_description.html;
        }

        return null;
    }, [data])

    const shouldRenderShortDescription = shortDescription ? <div className={classes.root}>
        <div className={classes.section}>
            <RichText content={shortDescription} />
        </div>
    </div> : null;

    return shouldRenderShortDescription;
};

export default ShortDescription;

ShortDescription.propTypes = {
    classes: shape({
        root: string,
        section: string
    }),
    productSku: string.isRequired
};
