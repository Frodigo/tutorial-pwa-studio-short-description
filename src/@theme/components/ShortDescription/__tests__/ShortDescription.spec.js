import React from 'react';
import { render, getByText } from '@testing-library/react';
import { useQuery } from '@apollo/client';

import ShortDescription from '../ShortDescription';

jest.mock('@apollo/client');

test('It renders component when the short description is filled in a product', () => {
    useQuery.mockReturnValue({
        data: {
            products: {
                items: [
                    {
                        uid: '1',
                        short_description: {
                            html: '<p>Lorem ipsum</p>'
                        }
                    }
                ]
            }
        }
    });

    const { container } = render(
        <ShortDescription productSku="abc"/>
    );

    expect(getByText(container, 'Lorem ipsum')).toBeDefined();
    expect(container).toMatchSnapshot();
});


test('It does not render when a product does not have the short description', () => {
    useQuery.mockReturnValue({
        data: {
            products: {
                items: [
                    {
                        uid: '1',
                        short_description: {
                            html: ''
                        }
                    }
                ]
            }
        }
    });

    const { container } = render(
        <ShortDescription productSku="abc"/>
    );

    expect(container.firstChild).toBeNull();
});
