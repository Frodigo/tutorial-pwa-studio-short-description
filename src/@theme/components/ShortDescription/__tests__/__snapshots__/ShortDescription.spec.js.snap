// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders component when the short description is filled in a product 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="section"
    >
      <div>
        <p>
          Lorem ipsum
        </p>
      </div>
    </div>
  </div>
</div>
`;
